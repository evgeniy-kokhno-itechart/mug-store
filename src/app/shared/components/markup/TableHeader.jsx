import React from 'react';
import { PropTypes } from 'prop-types';

const TableHeader = ({ columns }) => (
  <thead>
    <tr>
      {columns.map(column => (
        <th
          key={column.path || column.key}
          className={`p-1 p-md-2 p-lg-3 ${column.customClasses ? column.customClasses : ''}`}
        >
          {column.label}
        </th>
      ))}
    </tr>
  </thead>
);

TableHeader.propTypes = {
  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default TableHeader;
