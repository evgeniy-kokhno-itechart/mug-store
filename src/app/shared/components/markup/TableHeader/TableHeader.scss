@import 'Variables';

.infotable__header {
    border-bottom: 3px solid $color-gray-light-2;
}

.header-row__cell {
    text-align: left;
    padding: 1rem 0.5rem;
}