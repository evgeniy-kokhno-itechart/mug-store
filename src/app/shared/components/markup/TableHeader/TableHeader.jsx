import React from 'react';
import { PropTypes } from 'prop-types';
import './TableHeader.scss';

const TableHeader = ({ columns }) => (
  <thead className="info-table__header">
    <tr className="header-row">
      {columns.map(column => (
        <th
          key={column.path || column.key}
          className={`header-row__cell ${column.customClasses ? column.customClasses : ''}`}
        >
          {column.label}
        </th>
      ))}
    </tr>
  </thead>
);

TableHeader.propTypes = {
  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default TableHeader;
