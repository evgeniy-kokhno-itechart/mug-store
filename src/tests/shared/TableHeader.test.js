/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import TableHeader from '../../shared/markup-usage/TableHeader';

configure({ adapter: new Adapter() });

describe('<TableHeader />', () => {
  let TableHeaderWrapper;

  beforeEach(() => {
    TableHeaderWrapper = shallow(
      <TableHeader
        columns={[
          { path: 'id', label: 'ID' },
          { path: 'name', label: 'Name' },
          { path: 'description', label: 'Description', content: item => <span>{`Test Content ${item.description}`}</span> },
        ]}
      />,
    );
  });

  it('renders properly', () => {
    expect(TableHeaderWrapper).toMatchSnapshot();
  });
});
