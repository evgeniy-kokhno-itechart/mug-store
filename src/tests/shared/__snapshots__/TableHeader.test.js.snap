// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHeader /> renders properly 1`] = `
<thead>
  <tr>
    <th
      className="p-1 p-md-2 p-lg-3 "
      key="id"
    >
      ID
    </th>
    <th
      className="p-1 p-md-2 p-lg-3 "
      key="name"
    >
      Name
    </th>
    <th
      className="p-1 p-md-2 p-lg-3 "
      key="description"
    >
      Description
    </th>
  </tr>
</thead>
`;
